tools/build_langserver/langserver/utils_test.go

```diff
@@ -67,6 +72,54 @@ func TestEnsureURL(t *testing.T) {
 	assert.Equal(t, uri, lsp.DocumentURI("file://"+string(currentFile)))
 }
 
+func TestReadFile(t *testing.T) {
+	ctx := context.Background()
+	filepath := path.Join(core.RepoRoot, "tools/build_langserver/langserver/BUILD")
```

Comment:
whoa whoa whoa, this isn't right.

This file isn't one of the inputs to the test, and you're reading outside your test directory. Add a test file and pull it into the test using `data`.